### PR TITLE
Feature: Claim payment calculator

### PR DESCRIPTION
**Pull request by maxplt**
# Jira Issue: []

## What?
Make auto calculation for claim payment.

## Why?
For convenient setting amount.

## Video

https://user-images.githubusercontent.com/89198006/166883593-200368eb-ef67-4e56-803e-1c3441dad661.mov

 

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [ ] Codescouted
- [ ] Unit tests written
- [ ] Tested locally

### CODE REVIEW

**Comment by vonElfvin** [comment]
Super neat! I'd say there are rooms for improvement on the UX part though:
1. The total amount should not be editable
2. We should very clearly communicate what's going to happen (I.e. This will add a payment of amount X with deductible Y, type X to confirm)

Thoughts from Hunt?

**Comment by hunt-hedvig** [comment]
@vonElfvin @maxplt 
1. yea, I am probably overthinking about customising the final amount. Sorry Maxim, let's remove the edit btn and keep the net amount field not editable;)
2. update the text in the confirm payment dialog as `A payment of 500 SEK with 1500 SEK deductible will be added.  To perform the payment, type ‘500’ below to confirm.
`

**Comment by maxplt** [comment]
## Updates:

image

image

**Review by falklinus — APPROVED** [review]
🔥🔥

File: src/portals/hope/features/claims/claim-details/ClaimPayments/components/PaymentConfirmationModal.tsx [hunk 1]
```diff
@@ -169,11 +174,12 @@ export const PaymentConfirmationModal: React.FC<
           errors={errors}
         />
         <ButtonsGroup style={{ marginTop: '1em' }}>
-          <Button disabled={watch('confirmation') !== amount} type="submit">
+          <Button disabled={+watch('confirmation') !== +amount} type="submit">
```
Comment (rasmusguterstam) [inline src/portals/hope/features/claims/claim-details/ClaimPayments/components/PaymentConfirmationModal.tsx:177]: Why the `+` in front of `watch`?

**Review by rasmusguterstam — APPROVED** [review]

File: src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx [hunk 2]
```diff
@@ -163,39 +179,50 @@ export const ClaimPaymentForm: React.FC<{
...
+          affix={{ content: 'SEK' }}
+          errors={errors}
+          label="Amount"
+        />
+        <span style={{ marginBottom: '0.5rem' }}>-</span>
+        <Input
+          {...register('deductible')}
+          placeholder="Deductible"
+          label="Deductible"
+          step="any"
```
Comment (falklinus) [inline src/portals/hope/features/claims/claim-details/ClaimPayments/components/ClaimPaymentForm.tsx:197]: The design suggest we add the affix here as well.

**Pull request merged**